sel/src/Sel/Signing.hs

```diff
@@ -87,6 +126,32 @@ data SignedMessage = SignedMessage
   , signatureForeignPtr :: ForeignPtr CUChar
   }
 
+instance Eq SignedMessage where
+  (SignedMessage len1 msg1 sig1) == (SignedMessage len2 msg2 sig2) =
+    let msg1' = unsafeForeignPtrToPtr msg1
+        msg2' = unsafeForeignPtrToPtr msg2
+        sig1' = unsafeForeignPtrToPtr sig1
+        sig2' = unsafeForeignPtrToPtr sig2
+     in unsafeDupablePerformIO $
+          do m1 <- peek msg1'
+             m2 <- peek msg2'
+             s1 <- peek sig1'
+             s2 <- peek sig2'
+             return $ (len1, m1, s1) == (len2, m2, s2)
+
+instance Ord SignedMessage where
+  compare (SignedMessage len1 msg1 sig1) (SignedMessage len2 msg2 sig2) =
+    let msg1' = unsafeForeignPtrToPtr msg1
+        msg2' = unsafeForeignPtrToPtr msg2
+        sig1' = unsafeForeignPtrToPtr sig1
+        sig2' = unsafeForeignPtrToPtr sig2
+     in unsafeDupablePerformIO $
+          do m1 <- peek msg1'
+             m2 <- peek msg2'
+             s1 <- peek sig1'
+             s2 <- peek sig2'
+             return $ compare (len1, m1, s1) (len2, m2, s2)
```

Comment:
Same as for `HashKey`. Also, the 'repacking' into a tuple is not necessary here; just `&&` for `==` and `<>` for `compare`.